src/lambda/expression.rs

```diff
@@ -78,6 +85,15 @@ pub enum EvaluationError {
 
   #[error("APIValidationError: {0:?}")]
   APIValidationError(Vec<String>),
+
+  #[error("ConcatException: {0:?}")]
+  ConcatException(String),
+
+  #[error("IntersectionException: {0:?}")]
+  IntersectionException(String),
+
+  #[error("OperationFailed: {0:?}")]
+  OperationFailed(String),
```

Comment:
Generalize it to `ExprEvaluationError` don't need specialized errors for each type of expr operator.